### PR TITLE
Provider serves DNS proxy on Openvpn's virtual IP

### PR DESCRIPTION
**Pull request by Waldz**
Screen Shot 2019-08-27 at 15 19 31

### CODE REVIEW

File: services/openvpn/service/manager.go [hunk 1]
```diff
@@ -130,28 +134,40 @@ func (m *Manager) Serve(providerID identity.Identity) (err error) {
 		}
 	}()
 
-	if err = m.vpnServer.Start(); err != nil {
-		return
+	if err := m.startServer(m.vpnServer, stateChannel); err != nil {
+		return errors.Wrap(err, "failed to start Openvpn server")
+	}
+
+	m.dnsServer = dns.NewServer("10.8.0.1:53", dns.ResolveViaConfigured())
```
Comment (soffokl) [inline services/openvpn/service/manager.go]: "10.8.0.1/24" is just a default value, this subnet could be changed, we should respect it for DNS too.

Comment (Waldz) [reply services/openvpn/service/manager.go]: Improved

File: services/openvpn/service/manager.go [hunk 2]
```diff
@@ -115,12 +124,13 @@ func (m *Manager) Serve(providerID identity.Identity) (err error) {
...
 	vpnServerConfig := m.vpnServerConfigFactory(primitives, m.vpnServerPort)
-	m.vpnServer = m.vpnServerFactory(vpnServerConfig)
+	stateChannel := make(chan openvpn.State, 10)
+	m.vpnServer = m.vpnServerFactory(vpnServerConfig, stateChannel)
 
 	// register service port to which NATProxy will forward connects attempts to
 	m.natPinger.BindServicePort(openvpn_service.ServiceType, m.vpnServerPort)
 
-	log.Info(logPrefix, "starting openvpn server on port: ", m.vpnServerPort)
+	log.Info(logPrefix, "Starting openvpn server on port: ", m.vpnServerPort)
```
Comment (tadaskay) [inline services/openvpn/service/manager.go]: Case should be consistent, i.e. lowercase

Comment (Waldz) [reply services/openvpn/service/manager.go]: So logs most places we do with `log.Info("Uppercase ..")`.
Maybe You're mixing with `errors.New("lowercase ..")` which is Golang convention.

Comment (tadaskay) [reply services/openvpn/service/manager.go]: No @Waldz, I'm talking about the logging messages. They are not consistent, thus no, in most places, we don't do capitalization of the log messages. Quite often, the error is being logged as-is, resulting in a lowercase message as well. Perhaps it's a good addition for our devbook if we can agree on the choice?

**Pull request merged**